packages/autocomplete-js/src/getDefaultOptions.ts

```diff
@@ -165,6 +167,8 @@ export function getDefaultOptions<TItem extends BaseItem>(
           '--aa-detached-media-query'
         ),
       components: {
+        // @MAJOR Disallow registering components with the same name as default
+        // ones by merging the default components second.
```

Comment:
Alternative phrasing:

```suggestion
        // @MAJOR Deal with registering components with the same name as the default ones.
        // We could disallow registering these components by merging the default components second.
```

Reply:
Reworded here, wdyt? 5a5de5f3